### PR TITLE
Add epsilon for NaN prevention + test case to RetinaNet

### PR DESCRIPTION
**Pull request by LukeWood**
fixes https://github.com/keras-team/keras-cv/issues/1683

I actually took a `tf.cond()` based approach here to zero out sample weights instead.  I do this because I found that when we divide by normalizer, and normalizer is zero, the background classes all get absolutely massive values for class weight - causing massive gradients.

An alternative is perhaps to use a small value, i.e. 0.3~ for the class weights?  I think the ideal mask is something like: `tf.where(positive_mask, 0.1, 0.0)` where 0.1 is any tiny value.   Though, using all zeros is probably safer and easier.

### CODE REVIEW

**Review by ianstenbit — APPROVED** [review]
Awesome -- thank you Luke!

I'll try this same approach for YOLOV8

**Comment by LukeWood** [comment]
Looks like we need to upgrade tensorflow
